### PR TITLE
Fix backward compatible Pundit include

### PR DESCRIPTION
**Pull request by fbuys**
This commit fixes https://github.com/thoughtbot/administrate/issues/2378 By adding a conditional that includes Pundit::Authorization when it is defined else it includes Pundit.

This change ensures that we remain compatible with Pundit < 2.2.0

To test with Pundit "~> 2.1.0", we run:
```
 bundle exec appraisal pundit21 rspec spec/controllers/concerns/administrate/punditize_spec.rb   
 ```

See: https://github.com/thoughtbot/administrate/pull/2141
See: https://github.com/thoughtbot/administrate/pull/1068

### CODE REVIEW

File: app/controllers/concerns/administrate/punditize.rb [hunk 1]
```diff
@@ -2,7 +2,12 @@ module Administrate
   module Punditize
     if Object.const_defined?("Pundit")
       extend ActiveSupport::Concern
-      include Pundit::Authorization
+
+      if Pundit.const_defined?(:Authorization)
+        include Pundit::Authorization
+      else
+        include Pundit
```
Comment (fbuys) [inline app/controllers/concerns/administrate/punditize.rb:9]: This includes was deprecated in 2.2.0: https://github.com/varvet/pundit/blob/main/CHANGELOG.md#deprecated

File: spec/controllers/concerns/administrate/punditize_spec.rb [hunk 2]
```diff
@@ -2,15 +2,32 @@
 
 module Administrate
   RSpec.describe Punditize do
-    it "exists when 'Pundit' is not defined" do
-      unload_constants do
+    if Pundit.const_defined?(:Authorization)
```
Comment (fbuys) [inline spec/controllers/concerns/administrate/punditize_spec.rb:5]: Not sure how we feel about conditionals in spec files? Are there better alternatives?

**Review by fbuys — COMMENTED** [review]
Please let me know if I need to update any docs.

File: spec/controllers/concerns/administrate/punditize_spec.rb [hunk 3]
```diff
@@ -8,9 +8,26 @@ module Administrate
       end
     end
 
+    unless Pundit.const_defined?(:Authorization)
+      # Spec for pundit < 2.2
+      it "exists when Pundit::Authorization is not defined" do
+        reload_punditize
+        expect(Pundit.const_defined?(:Authorization)).to be_falsey
```
Comment (fbuys) [inline spec/controllers/concerns/administrate/punditize_spec.rb:15]: This assertion is somewhat redundant. The idea is to make it clear that Pundit::Authorization is not defined for punit < 2.2.

**Review by pablobm — APPROVED** [review]
Thank you!

**Comment by pablobm** [comment]
We appear to be having CI issues :sweat: I'll merge after we solve them.

**Comment by pablobm** [comment]
CI should be fixed in master now. Can you try rebase, please?

**Comment by fbuys** [comment]
> CI should be fixed in master now. Can you try rebase, please?

No problem, I have rebased and pushed again ✅

**Comment by pablobm** [comment]
Great stuff! Thank you.

**Pull request merged**